src/storage/QueryBaseProcessor.inl

```diff
@@ -274,21 +323,31 @@ void QueryBaseProcessor<REQ, RESP>::process(const cpp2::GetNeighborsRequest& req
     }
 
     // const auto& filter = req.get_filter();
-    std::for_each(req.get_parts().begin(), req.get_parts().end(), [&](auto& partV) {
-        auto partId = partV.first;
-        kvstore::ResultCode ret;
-        for (auto& vId : partV.second) {
-            VLOG(3) << "Process part " << partId << ", vertex " << vId;
-            ret = processVertex(partId, vId, tagContexts, edgeContext);
-            if (ret != kvstore::ResultCode::SUCCEEDED) {
-                break;
+    auto buckets = genBuckets(req);
+    std::vector<folly::Future<std::vector<OneVertexResp>>> results;
+    for (auto& bucket : buckets) {
+        results.emplace_back(asyncProcessBucket(std::move(bucket)));
+    }
+    folly::collectAll(results).via(executor_).thenTry([
+                     this,
+                     returnColumnsNum] (auto&& t) mutable {
+        CHECK(!t.hasException());
+        std::unordered_set<PartitionID> failedParts;
+        for (auto& bucketTry : t.value()) {
+            CHECK(!bucketTry.hasException());
+            for (auto& r : bucketTry.value()) {
+                auto& partId = std::get<0>(r);
+                auto& ret = std::get<2>(r);
+                if (ret != kvstore::ResultCode::SUCCEEDED
```

Comment:
if anything fails, go on or break?
do you need to mock a failure?

Reply:
Currently,  when one point failed,  the failure will be kept inside and go on to deal with the other points.  In the end,  we cluster the failures by part and return it back to client.